ibm/service/power/data_source_ibm_pi_virtual_serial_numbers.go

```diff
@@ -0,0 +1,96 @@
+// Copyright IBM Corp. 2024 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package power
+
+import (
+	"context"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+// Datasource to list Cloud Connections in a power instance
+func DataSourceIBMPIVirtualSerialNumbers() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMPIVirtualSerialNumbersRead,
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_InstanceID: {
+				Description: "ID of PVM instance to get virtual serial number attached to.",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+
+			// Attributes
+			Attr_VirtualSerialNumbers: {
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_Description: {
+							Computed:    true,
+							Description: "Description of virtual serial number.",
+							Type:        schema.TypeString,
+						},
+						Attr_InstanceID: {
```

Comment:
IMO this attr looks like redundant as it is same as `Arg_InstanceID` isn't?

Reply:
It will be only when `Arg_InstanceID` is supplied. However, it is needed whenever the argument is not supplied as it will be a list of virtual serial numbers with different values for InstanceID.